### ISSUE TITLE
There are some issues with the rules @stylistic/arrow-parens and @stylistic/space-before-function-paren when running in the ESLint plugin for VSCode

### ISSUE THREAD

**Issue by helloskynet** [eslint-stylistic/eslint-stylistic#209]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

with the rules @stylistic/arrow-parens and @stylistic/space-before-function-paren in ESLint Plugin for VSCode

Origin Code
```js
const m = async (id) => {
  return await Promise.resolve(id)
}
console.log(m)
```
Result
```js
const m = asyncid => {
  return await Promise.resolve(id)
}
console.log(m)
```

async (id) ------> asyncid

### Reproduction

https://stackblitz.com/~/github.com/helloskynet/eslint-p

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by kecrily** [comment]
I can't reproduce it. After running `eslint . --fix`, the code becomes

```js
const m = async id => {
  return await Promise.resolve(id)
}
console.log(m)
```

I think this is the expected behavior of the `'@stylistic/arrow-parens': ['error', 'as-needed']` config

**Comment by kecrily** [comment]
I think you should [go to the VS Code ESLint extension repository](https://github.com/microsoft/vscode-eslint) and submit the issue.
